### ISSUE TITLE
Request for additional mapping types and fields

### ISSUE THREAD

**Issue by numenor** [osm2vectortiles/osm2vectortiles#166]
I am working on conversion of an existing mapnik xml style to mapbox gl, based on osm2vectortiles data. While doing so, I encountered some mapping types and fields I am missing in osm2vectortiles to support the existing style. Below I am giving an overview of the most important ones. Is this list appropriate? Should I split it into several issues accoring to related tag, for instance? Is there any chance you might support these tags/values in future? To me it seems that most of these are a simple case of adding some rows to the mapping.yml and maybe data.yml files. But I do not know about the possible impact on tile sizes or other topics I might have overlooked.

Here are the issues I currently have (roughly in decreasing order of importance):

1. missing mappings for landuse / residential, landuse / commercial, landuse / retail, landuse / military, landuse / railway, landuse / greenfield, landuse / landfill, landuse / quarry, landuse / salt_pond, landuse / winter_sports. residential is main value for landuse, with > 3mio occurences

1. place / ...
important missing values: (important for low zoom, or with > 10.000 occurences): place / continent, place / country, place / island, place / islet
other important values for higher zoom levels: place / bay, place / farm, place / isolated_dwelling (4th most frequent value for place!)
other interesting values: place / archipelago, place / ocean, place / sea, place / state

1. capital, this is currently only a boolean feature, but most values in osm are integers with values up to 10, this would allow better distinction wrt. place importance

1. missing mapping for amenity / parking (around 2mio occurences)

1. natural / peak, >400,000 occurences

1. natural / ridge, natural / valley, important for labeling mountain topology (with around 35,000 and 5,000 occurences each)

1. replacement of landuse / beach (already present in osm2vectortiles, but occuring < 100 times) by natural / beach (ca. 100,000 occurences)

1. missing mappings for natural / rock, natural / bare_rock, natural / fell, natural / grassland, natural / heath, natural / riverbank, natural / scree

1. missing mappings for route / ferry, route / piste, route / ski, each several thousand occurrences

1. missing field tracktype for highway=track, with values tracktype / grade1, tracktype / grade2, tracktype / grade3, tracktype / grade4, tracktype / grade5

1. bicycle / designated, bicycle / yes, (maybe: horse / designated, ...) on highway=path, to allow more complete rendering of cycleways

1. missing mappings for leisure / marina, leisure / miniature_golf, leisure / track

1. missing mappings for military areas/buildings/pois: military / airfield, military / barracks, military / bunker, military / danger_area, military / naval_base, military / range

1. missing field usage for railways, to allow better distinction between different classes/importances: usage / branch, usage / highspeed, usage / industrial, usage / main, usage / military, usage / tourism

1. missing mapping of waterway / dock

1. missing mapping of aerialway / station (as poi and building shapes)

I would appreciate any indications about possible support for (any of) these in future!

**Comment by lukasmartinelli** [comment]
This is such a great listing @numenor!
We scheduled time for quality improvement but the big work would have been figuring out what is missing - exactly what you have done now!

*Did you previously base your Mapnik stylesheet on Mapbox Streets?*

Our layers, fields etc. all are directly based upon the decision to be compatible with the [Mapbox Streets ](https://www.mapbox.com/developers/vector-tiles/mapbox-streets-v6/#Overview).
With some of the issues you mentioned it is probably the case that we simply overlooked them when reverse engineering Mapbox Streets for others they are simply not there.

We do have the freedom to add additional data to the layers or introduce new layers for example peaks (as long as the tile size stays reasonable).
We also want to try to get the tile size under 500KB for the future (so people can use Mapbox Studio to style them).

*Did you find the no. of occurrences with taginfo?*

We will try to do a first assesment of those 16 issues listed and find out whether they would require a new layer or fit into an existing one.
We will do this assesment in this issue and after that we can split create new issues for the individual points listed.

**Comment by muesliq** [comment]
Great list! For prioritization purposes here the tags that are in my opinion missing most:

- landuse / residential
- place / * (to give islands, bays, etc. names)
- natural / peak
- natural / ridge, natural / valley
- route / ferry
- waterway / dock

**Comment by numenor** [comment]
@lukasmartinelli : Thanks for your feedback! Regarding your questions:

- Did you previously base your Mapnik stylesheet on Mapbox Streets?

No, I have a completely custom made style as mapnik xml, which I am trying to match as closely as possible with vector tile styling (taking advantage of their possibilities and Mapbox GL styling, of course). I do not have an "intermediate" version direcctly for Mapbox Streets. Thus, I cannot tell about whether/which of my suggestions are covered by Mapbox Streets. I was unsure of how much you want to diverge from / improve on Mapbox Streets, and thus how appropriate my suggestions are. I am glad though that you are considering these!

- Did you find the no. of occurrences with taginfo?

Exactly, the statistics by taginfo have been very valuable for prioritizing my suggestions ...

Just for completeness, another very important issue for me is already covered by #114 . I want to render pedestrian areas (a lot of central city areas are tagged highway=pedestrian, area=yes), but currently can only get the circumference instead of a fill (or alternatively, also all linear pedestrian ways are also artificially closed and filled, which would be worse).

**Comment by numenor** [comment]
**Regarding tiles maybe getting too large when adding further mappings/features:**

Did you consider creating more than one set of tiles for more specialized purposes (more diverse landcovers, or sports, or detailed POIs, or hiking/biking/skiing routes/pistes, ...)? In Mapbox Studio it is possible to add several sets of tiles, and depending on the purpose of a map, the appropriate ones could be chosen as data source. Such a modularized approach would thus still give reasonable amounts of data, while allowing more flexibility for map design. The cost, I guess, would be higher amounts of tile generation time and storage space ...

**Comment by klokan** [comment]
Anybody can use OSM2VectorTiles software stack (docker containers, etc) to generate a secondary vector tileset containing only extra tags or detailed POIs for a country or even whole world - which can be then by a tileserver combined with our official OSM2VectorTiles base maps during the request.

We plan to have the binary concatenation of several PBF vector tiles available in https://github.com/klokantech/tileserver-gl/ (which is also able to render GL JSON styles on the server side into raster tiles for clients not able to render directly with WebGL).

**Comment by numenor** [comment]
@klokan : Thanks to your publishing this project and the really nice documentation, I am aware of this possibility, and I am happy to have this as a fallback. On the other hand, I want to avoid duplicating effort for whatever of the mentioned mappings/features you consider useful for the tiles you generate. So I would like to clarify which of these could/should be incorporated into osm2vectortiles, and which you deem too specific and I will have to deal with in an additional own tileset.

**Comment by manuelroth** [comment]
@numenor thanks again for your comprehensive list. Below you can find the list with referenced commits if added or an explanation if not added. All commits can be found on the [mapping_improvements](https://github.com/osm2vectortiles/osm2vectortiles/commits/feature/mapping-improvements) branch. 

1. https://github.com/osm2vectortiles/osm2vectortiles/commit/98b4047dfb0bb493c321c7f051edeecfc90f445e Temporarily added, we will evaluate how this will affect the tile size and decide later if we really keep them
2. place={island, islet, isolated_dwelling} added with https://github.com/osm2vectortiles/osm2vectortiles/commit/2561827d71c1e8b75174530d2989660b99ffb5f1 .
place={country, continent, bay, ocean, sea, state} are not mapped, because we have custom lists for [countries](https://github.com/osm2vectortiles/osm2vectortiles/blob/master/src/import-external/countries.geojson), [states](https://github.com/osm2vectortiles/osm2vectortiles/blob/master/src/import-external/states.geojson) and [seas](https://github.com/osm2vectortiles/osm2vectortiles/blob/master/src/import-external/seas.geojson), therefore they are not listed in our import mapping, but are already present in our vector tiles.
3. Added the capital value with https://github.com/osm2vectortiles/osm2vectortiles/commit/2561827d71c1e8b75174530d2989660b99ffb5f1 place importance is distinguished with the scalerank and localrank, we will think about if we can incorporate this value into the scalerank calculation
4. https://github.com/osm2vectortiles/osm2vectortiles/commit/1ecf0b9409bcb1bb01c1e784f0df4965a6fe07d2
5. and 6. natural={peak, ridge, valley} will be part of new layer mountain_peak_label which is newly added by Mapbox Streets v7
7. https://github.com/osm2vectortiles/osm2vectortiles/commit/2e2eba178147df4b04d58705156ad8fd45889e17
8. https://github.com/osm2vectortiles/osm2vectortiles/commit/7cf2b128574df29b5f16f3c19ba60499a7704c1a and https://github.com/osm2vectortiles/osm2vectortiles/commit/62b589b74f1972cde524bed774ff4dacfa94fc7a
9. https://github.com/osm2vectortiles/osm2vectortiles/commit/79925ca9ef177dc9d4c212e5e55fc3ab9fcb0a6a
10. Mapbox Streets adds track:grade{1-5} as type for roads with class track
11. highway=path add attribute bicycle=designated and bicycle=yes for complete cycleway map => Mapbox Streets has special type for cycleway (general many different types for class=path)
12. https://github.com/osm2vectortiles/osm2vectortiles/commit/925e405c063c9911fcc3c94d3055c5e1dd66e4f9
13. Added with https://github.com/osm2vectortiles/osm2vectortiles/commit/98b4047dfb0bb493c321c7f051edeecfc90f445e landuse=military to layer landuse with this tag military={airfield, naval_base, training_area, barracks} are all covered
14. Add type for railways with usage(branch, highspeed, industrial, main, military, tourism) like for construction, track and service
15. https://github.com/osm2vectortiles/osm2vectortiles/commit/48a9f10c27a7d95becaa5c714644fe708be6c31d
16. https://github.com/osm2vectortiles/osm2vectortiles/commit/f26289c828fdc637bbf652c8706bc1c3318a52bc

The type classification in [Mapbox Streets v7](https://www.mapbox.com/developers/vector-tiles/mapbox-streets-v7/#road) has changed a lot. Applying the type classification will resolve section 9, 10 and 13. In a next step, I will align our [current mapping](https://github.com/osm2vectortiles/osm2vectortiles/blob/feature/mapping-improvements/src/import-osm/mapping.yml) and [source-style](https://github.com/osm2vectortiles/osm2vectortiles/blob/feature/mapping-improvements/open-streets.tm2source/data.yml) to all the changes the new Mapbox Streets v7 brought.

Before we can rerender the world again, the work on OSM Diff updates and the work on the mapping, source-style needs to be finished. As soon as we are ready we will inform you. If something is not clear, don't hesitate to ask me.

**Comment by manuelroth** [comment]
The work on Mapbox Streets v7 is tracked in this issue #204
